mesonbuild/mtest.py

```diff
@@ -129,6 +129,13 @@ def add_arguments(parser: argparse.ArgumentParser) -> None:
                         '"subprojname:" to run all tests defined by "subprojname".')
 
 
+def print_safe(s: str) -> None:
+    try:
+        print(s)
+    except UnicodeEncodeError:
+        s = s.encode('ascii', errors='replace').decode()
```

Comment:
is ascii really safe? Shouldn't we be using `sys.stdout.encoding`?

Reply:
That's pre-existing code that I am just moving around, I think the idea is to escape all non-7bit bytes. However, it should also `decode('ascii')`, I will make that a separate commit.